orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/JedisExecutionRepository.groovy

```diff
@@ -217,129 +235,225 @@ class JedisExecutionRepository implements ExecutionRepository {
+          def allowedExecutionStatuses = criteria.statuses*.toString() as Set<String>
+
+          def pipeline = jedis.pipelined()
+          def fetches = pipelineKeys.collect {
+            pipeline.hget("pipeline:${it}" as String, "status")
+          }
+          pipeline.sync()
 
-        fetches.eachWithIndex { Response<String> entry, int index ->
-          if (allowedExecutionStatuses.contains(entry.get())) {
-            filteredPipelineIds << pipelineKeys[index]
+          fetches.eachWithIndex { Response<String> entry, int index ->
+            if (allowedExecutionStatuses.contains(entry.get())) {
+              filteredPipelineIdsByJedis[jedisPool] << pipelineKeys[index]
+            }
           }
         }
       }
-
-      filteredPipelineIds = filteredPipelineIds.subList(0, Math.min(criteria.limit, filteredPipelineIds.size()))
     }
 
-    return retrieveObservable(Pipeline, executionsByPipelineKey(pipelineConfigId), new Func1<String, Iterable<String>>() {
-      @Override
-      Iterable<String> call(String key) {
-        withJedis { Jedis jedis ->
-          return filteredPipelineIds != null ? filteredPipelineIds : jedis.zrevrange(key, 0, (criteria.limit - 1))
+    /**
+     * Construct an observable that will retrieve pipelines from the primary redis
+     */
+    def currentPipelineIds = filteredPipelineIdsByJedis[jedisPool]
+    currentPipelineIds = currentPipelineIds.subList(0, Math.min(criteria.limit, currentPipelineIds.size()))
+
+    def currentObservable = retrieveObservable(
+      Pipeline,
+      executionsByPipelineKey(pipelineConfigId),
+      new Func1<String, Iterable<String>>() {
+        @Override
+        Iterable<String> call(String key) {
+          withJedis(jedisPool) { Jedis jedis ->
+            return criteria.statuses ? currentPipelineIds : jedis.zrevrange(key, 0, (criteria.limit - 1))
+          }
         }
-      }
-    }, queryByAppScheduler)
+      },
+      queryByAppScheduler,
+      jedisPool
+    )
+
+    if (jedisPoolPrevious.present) {
+      /**
+       * If configured, construct an observable the will retrieve pipelines from the secondary redis
+       */
+      def previousPipelineIds = filteredPipelineIdsByJedis[jedisPoolPrevious.get()]
+      previousPipelineIds = previousPipelineIds.subList(0, Math.min(criteria.limit, previousPipelineIds.size()))
+
+      def previousObservable = retrieveObservable(
+        Pipeline,
+        executionsByPipelineKey(pipelineConfigId),
+        new Func1<String, Iterable<String>>() {
+          @Override
+          Iterable<String> call(String key) {
+            withJedis(jedisPoolPrevious.get()) { Jedis jedis ->
+              return criteria.statuses ? previousPipelineIds : jedis.zrevrange(key, 0, (criteria.limit - 1))
+            }
+          }
+        },
+        queryByAppScheduler,
+        jedisPoolPrevious.get()
+      )
+
+      // merge primary + secondary observables
```

Comment:
The two observables are basically identical, could have a private method & avoid some duplication